app/controllers/api/v1/lookup_keys_controller.rb

```diff
@@ -58,9 +58,10 @@ def update
 
       def destroy
         if @lookup_key.type == "PuppetclassLookupKey"
-          render_error 'unprocessable_entity', :status => :unprocessable_entity
+          @lookup_key.errors.add(:base, 'Smart class parameters cannot be destroyed')
```

Comment:
I would generally agree, if it weren't for this being an obsolete controller.  It isn't a pattern that's likely to occur again.

I'd suggest just raising the error with `render_message` instead of messing with model errors, then add a functional test.